gap/prop.gi

```diff
@@ -38,58 +38,99 @@ D -> IsConnectedDigraph(D) and IsEmpty(ArticulationPoints(D)));
 InstallMethod(IsBridgelessDigraph, "for a digraph", [IsDigraph],
 D -> IsConnectedDigraph(D) and IsEmpty(Bridges(D)));
 
-InstallMethod(DIGRAPHS_IsMeetJoinSemilatticeDigraph, "for a homogeneous list",
-[IsHomogeneousList],
-function(nbs)
-  local i, j, k, n, x, len;
-
-  # The IsHomogenousList assumes the list <nbs> contains no or only empty lists
-  n := Length(nbs);
-  for i in [1 .. n] do
-    for j in [i + 1 .. n] do
-      if j in nbs[i] or i in nbs[j] then
-        continue;
-      fi;
-      x := Intersection(nbs[i], nbs[j]);
-      if IsEmpty(x) then
-        return false;
-      fi;
-      len := Length(x);
-      k := x[len];  # Check whether <k> is the meet of <i> and <j>
-      if Length(nbs[k]) < len then
+# The method below is based on Listing 11.9 of 'Free Lattices'
+# by Ralph Freese et. al.
+InstallMethod(DIGRAPHS_IsMeetJoinSemilatticeDigraph, "for a digraph and a bool",
+[IsDigraph, IsBool],
+function(D, join)
+  local P, U, ord, tab, S, N, i, x, T, l, q, z, y;
+
+  if not IsPartialOrderDigraph(D) then
+    return false;
+  fi;
+
+  D   := DigraphImmutableCopyIfMutable(D);
```

Comment:
```suggestion
  D   := DigraphMutable(D);
```

Reply:
I think this is replaced by what we discussed later on